lib/KDateRange/KDateInput.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+
+  <div>
+    <fieldset :aria-labelledby="legendText" class="date-input-fieldset" aria-describedby="date-desc" aria-live="polite">
+      <KTextBox
+        :ref="inputRef"
+        :value="value"
+        type="text"
+        :label="legendText"
+        autoComplete="off"
+        :invalid="Boolean(errorMessage)"
+        :showInvalidText="Boolean(errorMessage)"
+        :invalidText="errorMessage"
+        @input="handleInput"
+      />  
+      <input type="hidden" name="date" :value="valueAsDate" data-test="valueAsDate">
+      <span class="k-date-vhidden">
+        <span v-if="valueAsDate" id="date-desc">
+          {{ dateDescription }}
+        </span>
+      </span>
+    </fieldset>
+  </div>
+
+</template>
+
+
+<script>
+
+  import { startOfDay } from 'date-fns';
+  import KTextBox from '../KTextbox';
+  import { DATE_FMT } from './validationConstants';
+
+  export default {
+    name: 'KDateInput',
+    components: {
+      KTextBox,
+    },
+    props: {
+      dateLocale: {
+        type: String,
+        required: true,
+      },
+      inputRef: {
+        type: String,
+        default: null,
+      },
+      legendText: {
+        type: String,
+        required: true,
+      },
+      errorMessage: {
+        type: [String, Boolean],
+        default: false,
+      },
+      value: {
+        type: String,
+        default: DATE_FMT,
+      },
+    },
+    computed: {
+      valueAsDate() {
+        if (this.value) {
+          return this.createDate(this.value);
+        }
+        return '';
+      },
+      dateDescription() {
+        if (
+          Object.prototype.toString.call(this.valueAsDate) === '[object Date]' &&
+          isNaN(this.valueAsDate)
+        ) {
+          return ' ';
+        } else {
```

Comment:
Since the prior `if` block returns, you can safely extract this from the `else`, like you did with `valueAsDate`. Makes the code a little cleaner